test/Confluent.Kafka.UnitTests/Timestamp.cs

```diff
@@ -64,5 +64,30 @@ public void Conversion()
             Assert.Equal(ts, ts2);
             Assert.Equal(ts2.Kind, DateTimeKind.Utc);
         }
+
+        [Fact]
+        public void Rounding()
+        {
```

Comment:
you could add a test to subtract a tick from a millisecond aligned datetime and see that it rounds down (optional - only if you feel like it).